### ISSUE TITLE
Example of APIG and Lambda integration

### ISSUE THREAD

**Issue by damluar** [awslabs/serverless-application-model#242]
In the example `api_swagger_cors` you connect Lambda to events from APIG in the following way:
```
Events:
        ProxyApiRoot:
          Type: Api
          Properties:
            RestApiId: !Ref ApiGatewayApi
            Path: /
            Method: ANY
```

But you also specify the mapping in the swagger definition (only POST for whatever reason):
```
/:
  ...
  uri: arn:aws:apigateway:<<region>>:lambda:path/2015-03-31/functions/arn:aws:lambda:<<region>>:<<accountId>>:function:${stageVariables.LambdaFunctionName}/invocations
  passthroughBehavior: when_no_match
  httpMethod: POST
```
Is it a duplication? If not, what is the difference?

**Comment by Nr18** [comment]
The `httpMethod` maps tp http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-apitgateway-method-integration.html#cfn-apigateway-method-integration-integrationhttpmethod so it is they method how the Lambda function is invoked.

**Comment by damluar** [comment]
Hi Joris, I am afraid I don't understand your answer. Do you want to say that we need to use both of the above mentioned definitions or would one be sufficient?

**Comment by Nr18** [comment]
No problem, let me try to clarify so you have 2 method statements:

1. The way that the API Gateway is invoked
2. The way that your integration is invoked

A Lambda function requires a POST while the request on the API Gateway could be a GET, so for instance:

- You have a resource /my/path 
- You place a GET method under that resource you could: 
  - Invoke a Lambda function and the integration method should be POST.
  - Proxy to an existing endpoint (depending on the API you are proxying) you can use a GET method for your integration.

Does that makes sense?

**Comment by sanathkr** [comment]
Yeah @Nr18  is correct. 

Your client -- HTTP GET/POST/DELETE/ANY/* --> API Endpoint --- POST (internally called by APIGW) --> Lambda function

The POST is an implementation detail and not relevant to your clients.

**Comment by sanathkr** [comment]
Closing this issue, since it has been answered. Feel free to re-open if you have comments
